ground/src/main/java/com/google/android/ground/ui/datacollection/components/TaskViewFactory.kt

```diff
@@ -26,24 +26,11 @@ import com.google.android.ground.databinding.TaskFragWithoutHeaderBinding
 /** Factory for creating a [TaskView]. */
 object TaskViewFactory {
 
-  fun createWithHeader(layoutInflater: LayoutInflater): TaskView {
-    val binding = TaskFragWithHeaderBinding.inflate(layoutInflater)
-    return TaskViewWithHeader(binding)
-  }
+  fun createWithHeader(layoutInflater: LayoutInflater): TaskView =
+    TaskViewWithHeader(TaskFragWithHeaderBinding.inflate(layoutInflater))
 
-  fun createWithoutHeader(
-    layoutInflater: LayoutInflater,
-    @DrawableRes iconResId: Int? = null,
-    @StringRes labelResId: Int? = null
-  ): TaskView {
-    val binding = TaskFragWithoutHeaderBinding.inflate(layoutInflater)
-    iconResId?.let {
-      val drawable = AppCompatResources.getDrawable(layoutInflater.context, it)
-      binding.headerIcon.setImageDrawable(drawable)
-    }
-    labelResId?.let { binding.headerLabel.setText(labelResId) }
-    return TaskViewWithoutHeader(binding)
-  }
+  fun createWithoutHeader(layoutInflater: LayoutInflater): TaskView =
+    TaskViewWithoutHeader(TaskFragWithoutHeaderBinding.inflate(layoutInflater))
```

Comment:
Do we need a new class `TaskViewWithCollapsibleHeader`?

[(Figma mocks)](https://www.figma.com/file/r2uGtyOiWoDa29za7DNuUc/Ground-Acacia-UX?node-id=15%3A11184&mode=dev)

Reply:
Which tasks are you referring to from the mocks? I don't think I see any with collapsible headers

Reply:
"Take a photo" task mock has 4 screenshots and in the fourth one the header is collapsed.

Reply:
Oh yea I think you're right.  I'll plan to address the collapsing on scroll in a follow up PR.

Reply:
SG!